yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -59,43 +58,46 @@ private[yarn] class ExecutorRunnable(
 
   var rpc: YarnRPC = YarnRPC.create(conf)
   var nmClient: NMClient = _
-  val yarnConf: YarnConfiguration = new YarnConfiguration(conf)
-  lazy val env = prepareEnvironment(container)
 
   def run(): Unit = {
-    logInfo("Starting Executor Container")
+    logDebug("Starting Executor Container")
     nmClient = NMClient.createNMClient()
-    nmClient.init(yarnConf)
+    nmClient.init(conf)
     nmClient.start()
     startContainer()
   }
 
-  def startContainer(): java.util.Map[String, ByteBuffer] = {
-    logInfo("Setting up ContainerLaunchContext")
+  def launchContextDebugInfo(): String = {
+    val commands = prepareCommand()
```

Comment:
I thought about adding a boolean "has the context been logged?" to control when to log. But then it would contain information about the first executor only, and this allows be to put placeholders for executor id and hostname, for example. So the log message looks a little better even if the code is not optimal.

As for the command / env, the command is mostly the same, aside from the exec id and hostname; so every launcher would still need to do some customization.